lib/rules/no-get-with-default.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const types = require('../utils/types');
+
+function makeErrorMessageForGetWithDefault(property) {
+  return `Use \`get\` and \`||\` instead of \`getWithDefault\` for property \`${property}\``;
+}
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Use the || operator instead of getWithDefault for more expected behaviors',
+      category: 'Best Practices',
+      recommended: false,
+      octane: true,
```

Comment:
What's the reasoning for this to be part of the `octane` config? CC: @patocallaghan

Reply:
I can see a reason why it would be added if we already have the `no-get` rule as part of the octane config. This is just promoting the usage of native javascript over classic Ember APIs.

Reply:
Sounds good to me!